libnest/src/package/mod.rs

```diff
@@ -1,3 +1,109 @@
 //! Provides types and functions to interact with all kinds of packages: available ones, installed ones etc.
 
+pub mod identification;
 pub mod manifest;
+
+/// A package's kind.
+///
+/// All entities called 'package' may not represent the same thing.
+/// Some are actual binaries or libraries like one may expect ('effective' packages), but
+/// others may be entirely empty, used only to name a list of dependencies ('virtual' packages).
+///
+/// The `Kind` enum is used to differentiate those packages and speed up their installation process.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub enum Kind {
+    /// The package contains some installable data.
+    Effective,
+    /// The package doesn't contain any data.
+    Virtual,
+}
+
+impl Default for Kind {
+    fn default() -> Kind {
+        Kind::Effective
+    }
+}
+
+/// A package's name.
+///
+/// A `PackageName` can be casted to an `&str` and ensures, when created, that the underlying string matches
+/// the expectations of what a package's name should look like.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug, Default)]
+pub struct PackageName(String);
+
+impl PackageName {
+    /// Create a [`PackageName`] from a String
+    pub fn from(name: String) -> Self {
```

Comment:
Ideally, such functions should perform a verification to make sure the package name is valid (Remember, a package name must match `^[a-z0-9\-]+$`), and throw an error otherwise. 

The same thing applies for the other structures below.